src/couch/src/couch_server.erl

```diff
@@ -301,22 +306,22 @@ terminate(Reason, Srv) ->
         if Db == undefined -> ok; true ->
             couch_util:shutdown_sync(couch_db:get_pid(Db))
         end
-    end, nil, couch_dbs),
+    end, nil, couch_dbs(Srv)),
     ok.
 
 handle_config_change("couchdb", "database_dir", _, _, _) ->
     exit(whereis(couch_server), config_change),
     remove_handler;
-handle_config_change("couchdb", "update_lru_on_read", "true", _, _) ->
-    {ok, gen_server:call(couch_server,{set_update_lru_on_read,true})};
-handle_config_change("couchdb", "update_lru_on_read", _, _, _) ->
-    {ok, gen_server:call(couch_server,{set_update_lru_on_read,false})};
-handle_config_change("couchdb", "max_dbs_open", Max, _, _) when is_list(Max) ->
-    {ok, gen_server:call(couch_server,{set_max_dbs_open,list_to_integer(Max)})};
-handle_config_change("couchdb", "max_dbs_open", _, _, _) ->
-    {ok, gen_server:call(couch_server,{set_max_dbs_open,?MAX_DBS_OPEN})};
-handle_config_change("couchdb_engines", _, _, _, _) ->
-    {ok, gen_server:call(couch_server, reload_engines)};
+handle_config_change("couchdb", "update_lru_on_read", "true", _, N) ->
+    {ok, gen_server:call(couch_server(N),{set_update_lru_on_read,true})};
+handle_config_change("couchdb", "update_lru_on_read", _, _, N) ->
+    {ok, gen_server:call(couch_server(N),{set_update_lru_on_read,false})};
+handle_config_change("couchdb", "max_dbs_open", Max, _, N) when is_list(Max) ->
+    {ok, gen_server:call(couch_server(N),{set_max_dbs_open,list_to_integer(Max)})};
+handle_config_change("couchdb", "max_dbs_open", _, _, N) ->
+    {ok, gen_server:call(couch_server(N),{set_max_dbs_open,?MAX_DBS_OPEN})};
+handle_config_change("couchdb_engines", _, _, _, N) ->
+    {ok, gen_server:call(couch_server(N), reload_engines)};
```

Comment:
Hmmm... do we want to trigger the config change callbacks `N` times now? The current approach keeps the code simpler, and I'm being a bit pedantic here, but it might be worth making a dedicated config listener pid that loops over the couch_server pids to trigger updates.

Reply:
Each of the couch_servers get the callback and do this once each.

Reply:
Sure, I just mean it's `N` numbers of config listeners for the same config changes, as each couch_server subscribes to the changes. Like I said, being pedantic here, but we could make a `couch_server_config_listener` that subscribes once and then updates all the `couch_servers` in a loop.